dask/dataframe/io/csv.py

```diff
@@ -942,7 +942,7 @@ def to_csv(
         first_file = open_file(filename, mode=mode, **file_options)
         value = to_csv_chunk(dfs[0], first_file, **kwargs)
         append_mode = mode if "a" in mode else mode + "a"
-        append_mode = append_mode.replace("w", "")
+        append_mode = mode.replace("w", "").replace("x", "")
```

Comment:
Sorry, I messed up the merge with `main`.
```suggestion
        append_mode = append_mode.replace("w", "").replace("x", "")
```